src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2113,7 +2138,12 @@ mgp_error mgp_graph_detach_delete_vertex(struct mgp_graph *graph, mgp_vertex *ve
       }
     }
 
-    auto *trigger_ctx_collector = graph->ctx->trigger_context_collector;
+    auto &ctx = graph->ctx;
+
+    ctx->execution_stats[query::ExecutionStats::Key::DELETED_NODES] += 1;
+    ctx->execution_stats[query::ExecutionStats::Key::DELETED_EDGES] += (*result)->second.size();
```

Comment:
At this point `result` can be `std::nullopt`, therefore I would push this after the following `if` statement after which `result` holds a valid value.